### PR TITLE
Add Sprint2 demo scene, player controls, interactables, and riddles demo 

### PR DESCRIPTION
**Pull request by travis-bittel**
Merge after all other PRs

### CODE REVIEW

File: Gaming The Syllabus/Assets/Player.cs [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+    }
+
+    private void FixedUpdate()
+    {
+        if (CanMove)
+        {
+            rigidBody.MovePosition((Vector2)transform.position + movementDirection * movementSpeed * Time.deltaTime);
+        }
+    }
+
```
Comment (MatthewEpperson) [inline Gaming The Syllabus/Assets/Player.cs:31]: Much better movement system than our initial implementation. Well done!

File: Gaming The Syllabus/Assets/RiddleDisplayer.cs [hunk 2]
```diff
@@ -0,0 +1,13 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class RiddleDisplayer : MonoBehaviour
+{
+    [SerializeField] private int indexOfRiddleToDisplay;
+
```
Comment (MatthewEpperson) [inline Gaming The Syllabus/Assets/RiddleDisplayer.cs:8]: Smart choice on making this private and a serializefield.

File: Gaming The Syllabus/Assets/Scripts/Combat/CombatUnit.cs [hunk 3]
```diff
@@ -20,9 +20,10 @@ public class CombatUnit : IXmlSerializable
 
     public bool IsAlive { get; private set; } = true;
 
-    public void PerformAction(int actionIndex) 
+    public void PerformAction(int actionIndex)
     {
-        if (IsAlive) {
+        if (IsAlive)
+        {
```
Comment (MatthewEpperson) [inline Gaming The Syllabus/Assets/Scripts/Combat/CombatUnit.cs:26]: I like how you like to keep our brackets consistent. Consistency is always good.

File: Gaming The Syllabus/Assets/Scripts/EventBasedInteractable.cs [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+        }
+    }
+
+    private void OnTriggerExit2D(Collider2D collision)
+    {
+        if (collision.CompareTag("Player"))
+        {
+            Player.RemoveInteractable(this);
+        }
+    }
```
Comment (MatthewEpperson) [inline Gaming The Syllabus/Assets/Scripts/EventBasedInteractable.cs:34]: I would have gone about this most likely in a more complex and complicated way. Glad you were able to accomplish it much more simply lol

File: Gaming The Syllabus/Assets/Scripts/IInteractable.cs [hunk 5]
```diff
@@ -0,0 +1,10 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public interface IInteractable
+{
+    public int InteractionPriority { get; }
+
+    public void Interact();
+}
```
Comment (MatthewEpperson) [inline Gaming The Syllabus/Assets/Scripts/IInteractable.cs:10]: Great choice on making this an interface. Should make further development much easier.

File: Gaming The Syllabus/Assets/Scripts/Riddles/UI/AnswerSubmissionManager.cs [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+                _instance.selectedAnswers.Clear();
+            }
+            _instance.selectedAnswers.Add(element);
+            
+        }
+        else if (_instance.selectedAnswers.Contains(element))
+        {
+            _instance.selectedAnswers.Remove(element);
+        }
+    }
```
Comment (MatthewEpperson) [inline Gaming The Syllabus/Assets/Scripts/Riddles/UI/AnswerSubmissionManager.cs:33]: I'm assuming this sets the answer the player chooses for a riddle.

File: Gaming The Syllabus/Assets/Scripts/Riddles/UI/RiddleAnswerButton.cs [hunk 7]
```diff
@@ -0,0 +1,36 @@
...
+    public bool IsSelectedAnswer { get; set; }
+
+    private RiddleAnswer associatedAnswer;
+
+    [SerializeField] private TextMeshProUGUI text;
+
+    public void OnClick()
+    {
+        AnswerSubmissionManager.SetAnswerElementSelected(this, true, deselectAllOthers: _associatedRiddle.TypeOfAnswer == AnswerType.SelectOne);
+    }
```
Comment (MatthewEpperson) [inline Gaming The Syllabus/Assets/Scripts/Riddles/UI/RiddleAnswerButton.cs:23]: Ahh yes okay. So this checks for which answer the player clicks on. Got it.

File: Gaming The Syllabus/Assets/Scripts/Riddles/UI/RiddleAnswerUIElement.cs [hunk 8]
```diff
@@ -0,0 +1,27 @@
...
+    public abstract RiddleAnswer GetAnswer();
+
+    /// <summary>
+    /// Do any necessary setup for the element here, such as setting its text component if applicable.
+    /// </summary>
+    /// <param name="associatedAnswer"></param>
+    public abstract void Initialize(Riddle associatedRiddle, RiddleAnswer associatedAnswer); 
+
+    public abstract Riddle AssociatedRiddle { get; }
+}
```
Comment (MatthewEpperson) [inline Gaming The Syllabus/Assets/Scripts/Riddles/UI/RiddleAnswerUIElement.cs:27]: Another great use of utilizing interfaces. Great job!

File: Gaming The Syllabus/Assets/Scripts/EventBasedInteractable.cs (hunk 4 above)
Comment (travis-bittel) [reply Gaming The Syllabus/Assets/Scripts/EventBasedInteractable.cs:34]: I have done this probably 50 different ways on 50 different projects and hated every single way I've done it except this one lol... have spent way too much time fighting with this sort of stuff

File: Gaming The Syllabus/Assets/AnswerSubmissionManager.cs [hunk 9]
```diff
@@ -0,0 +1,39 @@
...
+            {
+                _instance.selectedAnswers.Clear();
+            }
+            _instance.selectedAnswers.Add(element);
+            
+        }
+        else if (_instance.selectedAnswers.Contains(element))
+        {
+            _instance.selectedAnswers.Remove(element);
+        }
```
Comment (Isildur512) [inline Gaming The Syllabus/Assets/AnswerSubmissionManager.cs]: Smart way to optimize this!

**Pull request merged**